package com.example.ch05starter

import androidx.compose.foundation.background
import androidx.compose.foundation.layout.Arrangement
import androidx.compose.foundation.layout.Box
import androidx.compose.foundation.layout.PaddingValues
import androidx.compose.foundation.layout.aspectRatio
import androidx.compose.foundation.layout.padding
import androidx.compose.foundation.lazy.grid.GridCells
import androidx.compose.foundation.lazy.grid.LazyVerticalGrid
import androidx.compose.foundation.lazy.grid.items
import androidx.compose.material3.ExperimentalMaterial3Api
import androidx.compose.material3.Scaffold
import androidx.compose.material3.Text
import androidx.compose.material3.TopAppBar
import androidx.compose.runtime.Composable
import androidx.compose.ui.Modifier
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.tooling.preview.Preview
import androidx.compose.ui.unit.dp


data class Photo(
    val id: Int,
    val color: Color
)


val dummyPhotos = (1..21).map { i ->
    Photo(
        id = i,
        color = Color(
            red = (i * 37 % 256) / 255f,
            green = (i * 91 % 256) / 255f,
            blue = (i * 53 % 256) / 255f
        )
    )
}


@OptIn(ExperimentalMaterial3Api::class)
@Composable
fun ExploreScreen() {

    Scaffold(
        topBar = {
            TopAppBar(
                title = {
                    Text("Eksplorasi")
                }
            )
        }
    ) { paddingValues ->

        LazyVerticalGrid(
            modifier = Modifier.padding(paddingValues),
            columns = GridCells.Fixed(2),
            contentPadding = PaddingValues(8.dp),
            verticalArrangement = Arrangement.spacedBy(8.dp),

            horizontalArrangement = Arrangement.spacedBy(8.dp)
        ) {

            items(
                items = dummyPhotos,
                key = { photo -> photo.id }
            ) { photo ->
                Box(
                    modifier = Modifier
                        .aspectRatio(1f)
                        .background(photo.color)
                )
            }
        }
    }
}


@Preview(
    showBackground = true,
    showSystemUi = true
)
@Composable
fun ExploreScreenPreview() {
    ExploreScreen()
}